packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -347,28 +413,129 @@ async function selectPlatformAndExecute(
     });
 
     if (targetName === undefined) {
-      vscode.window.showInformationMessage(commandCancelledMessage);
-      return;
+      // user cancelled operation
+      return undefined;
     }
   }
 
-  await startServer(false, componentName, startTime);
-
-  // New target device entered
-  if (targetName !== '') {
+  // new target device entered
+  if (targetName && targetName !== '') {
     PreviewService.instance.updateRememberedDevice(
       platformSelection.platformName,
       targetName
     );
+
     target = targetName;
   }
 
-  const previewCommand = new SfdxCommandBuilder()
+  return target;
+}
+
+/**
+ * Prompts the user to select an app to preview the LWC on. Defaults to browser.
+ *
+ * @param platformSelection the selected platform
+ * @param configFile path to a config file
+ * @returns the name of the selected device or undefined if user cancels selection.
+ */
+async function selectTargetApp(
+  platformSelection: PreviewQuickPickItem,
+  configFile: string | undefined
+): Promise<string | undefined> {
+  let targetApp: string | undefined = 'browser';
+  const options: vscode.QuickPickItem[] = [];
+
+  if (configFile === undefined || fs.existsSync(configFile) === false) {
+    return targetApp;
+  }
+
+  try {
+    const fileContent = fs.readFileSync(configFile, 'utf8');
+    const json = JSON.parse(fileContent);
+    const appDefinitionsForSelectedPlatform =
+      platformSelection.id === PreviewPlatformType.Android
+        ? json.apps.android
+        : json.apps.ios;
+
+    const apps = Array.from<any>(appDefinitionsForSelectedPlatform);
+
+    apps.forEach(app => {
+      const label: string = app.name;
+      const detail: string = app.id;
+      options.push({ label, detail });
+    });
+  } catch {
+    // silengtly fail and default to previewing on browser
+  }
+
+  // if there are any devices available, show a pick list.
+  if (options.length > 0) {
+    options.unshift({
```

Comment:
Can we make an explicit `browserItem` variable out of this, and then check its equality to `selectedItem` [below](https://github.com/forcedotcom/salesforcedx-vscode/pull/2585/files#diff-785bdec339e64f40207b4e9dcae14dd4R486), instead of comparing the label? Identifying a component by its label feels fragile.